x-pack/plugins/ml/server/models/job_service/jobs.ts

```diff
@@ -271,6 +272,22 @@ export function jobsProvider(
     return jobs;
   }
 
+  async function getJobsWithGeo(): Promise<string[]> {
+    const { body } = await mlClient.getJobs<MlJobsResponse>();
+
+    const geoJobs: string[] = [];
+
+    if (body.count && body.count > 0) {
+      body.jobs.forEach((job) => {
+        if (isJobWithGeoData(job)) {
+          geoJobs.push(job.job_id);
+        }
+      });
+    }
+
+    return geoJobs;
+  }
```

Comment:
I think this method should be called `getJobIdsWithGeo`, because it provides job ids. It also can be simplified.
```suggestion
  async function getJobIdsWithGeo(): Promise<string[]> {
    const { body } = await mlClient.getJobs<MlJobsResponse>();
    return body.jobs.filter(isJobWithGeoData).map(job => job.id);
  }
```

Reply:
Simplified in e121d6724193fd765c79cc6fb03af95cb4377510